AEPMessaging/Sources/Messaging.swift

```diff
@@ -218,59 +387,40 @@ public class Messaging: NSObject, Extension {
-            Log.trace(label: MessagingConstants.LOG_TAG, "The personalization:decisions response contains InApp message definitions.")
-            rulesEngine.launchRulesEngine.loadRules(inAppRules, clearExisting: clearExistingRules)
-        }
-
-        if let feedItemRules = rules[InboundType.feed] {
-            Log.trace(label: MessagingConstants.LOG_TAG, "The personalization:decisions response contains feed message definitions.")
-            feedRulesEngine.launchRulesEngine.loadRules(feedItemRules, clearExisting: clearExistingRules)
-            if let inboundMessages = feedRulesEngine.evaluate(event: event) {
-                updateInboundMessages(inboundMessages, surfaces: requestedSurfaces)
-            }
+        
+        Log.trace(label: MessagingConstants.LOG_TAG, "Processing propositions from personalization:decisions network response for event '\(requestEventId)'.")
+        updateInProgressPropositionsWith(event)
+    }
+    
+    private func updateInProgressPropositionsWith(_ event: Event) {
+        guard let requestingEventId = event.requestEventId else {
+            Log.trace(label: MessagingConstants.LOG_TAG, "Ignoring personalization:decisions response with no requesting Event ID.")
+            return
         }
-
-        let requestedPropositions = retrievePropositions(surfaces: requestedSurfaces)
-        guard !requestedPropositions.isEmpty else {
-            Log.trace(label: MessagingConstants.LOG_TAG, "Not dispatching a notification event, personalization:decisions response does not contain propositions.")
+        guard let eventPropositions = event.payload else {
+            Log.trace(label: MessagingConstants.LOG_TAG, "Ignoring personalization:decisions response with no propositions.")
             return
         }
-
-        // dispatch an event with the propositions received from the remote
-        let eventData = [MessagingConstants.Event.Data.Key.PROPOSITIONS: requestedPropositions.flatMap { $0.value }].asDictionary()
-
-        let event = Event(name: MessagingConstants.Event.Name.MESSAGE_PROPOSITIONS_NOTIFICATION,
-                          type: EventType.messaging,
-                          source: EventSource.notification,
-                          data: eventData)
-        dispatch(event: event)
+        
+        // loop through propositions for this event and add them to existing props by surface
+        for proposition in eventPropositions {
+            let surface = Surface(uri: proposition.scope)
+            var newPropositions = inProgressPropositions[surface] ?? []
+            Log.trace(label: MessagingConstants.LOG_TAG, "Adding proposition for surface '\(surface.uri)' for event '\(requestingEventId)'.")
+            newPropositions.append(proposition)
+            Log.trace(label: MessagingConstants.LOG_TAG, "There are now \(newPropositions.count) proposition(s) for this surface.")
+            inProgressPropositions[surface] = newPropositions
```

Comment:
yeah - i did it this way before because the logging was nice with the idea of handling multiple in-flight requests at once, but no longer necessary now